### ISSUE TITLE
[BUG]: GroupKfold no longer working in v3.x

### ISSUE THREAD

**Issue by ninjit** [pycaret/pycaret#3016]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

This maybe related to other groupkfold issues opened:
https://github.com/pycaret/pycaret/issues?q=is%3Aissue+is%3Aopen+groupkfold
Such as  #2259 

The blank output from compare_models() is likely due to underlying create_models failing on the groupkfolds

For my purposes, I've been able to get this working by explicitly removing the Group_ID column from the dataset, and passing it independently to setup()

However this no longer works in v3.rc

### Reproducible Example

```python
import pandas as pd
from pycaret.datasets import get_data
dataset = get_data('credit')

#check the shape of data
dataset.shape

dataset["EDUCATION"] = pd.Categorical(dataset["EDUCATION"])

data = dataset.sample(frac=0.95, random_state=786)
data_unseen = dataset.drop(data.index)
data.reset_index(inplace=True, drop=True)
data_unseen.reset_index(inplace=True, drop=True)
print('Data for Modeling: ' + str(data.shape))
print('Unseen Data For Predictions: ' + str(data_unseen.shape))

# get groups and remove from dataset
# !pycaret specify as column label seems broken, as group column is kept within training data and not separated
data_groups = data["EDUCATION"]
data.drop("EDUCATION", axis=1, inplace=True)

data_unseen_groups = data_unseen["EDUCATION"]
data_unseen.drop("EDUCATION", axis=1, inplace=True)

from pycaret.classification import *
from pycaret import __version__
__version__

exp_clf101	=	setup(data = data, target = 'default', session_id = 123, use_gpu = False,
					fold=5, fold_strategy='groupkfold', fold_groups=data_groups,# silent=True,
					) 

best = compare_models()

lr = create_model('lr')
```

### Expected Behavior

compare_models() return table of performance results
create_model() returns kfold cross-validation results for specified model type

### Actual Results

```python-traceback
compare_models() runs quickly and is empty

create_model(...) results in the follow error

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In [9], line 1
----> 1 lr = create_model('lr')

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/pycaret/utils/generic.py:950, in check_if_global_is_not_none.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    948     if globals_d[name] is None:
    949         raise ValueError(message)
--> 950 return func(*args, **kwargs)

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/pycaret/classification/functional.py:1004, in create_model(estimator, fold, round, cross_validation, fit_kwargs, groups, probability_threshold, experiment_custom_tags, engine, verbose, return_train_score, **kwargs)
    874 @check_if_global_is_not_none(globals(), _CURRENT_EXPERIMENT_DECORATOR_DICT)
    875 def create_model(
    876     estimator: Union[str, Any],
   (...)
    887     **kwargs,
    888 ) -> Any:
    890     """
    891     This function trains and evaluates the performance of a given estimator
    892     using cross validation. The output of this function is a score grid with
   (...)
   1001 
   1002     """
-> 1004     return _CURRENT_EXPERIMENT.create_model(
   1005         estimator=estimator,
   1006         fold=fold,
   1007         round=round,
   1008         cross_validation=cross_validation,
   1009         fit_kwargs=fit_kwargs,
   1010         groups=groups,
   1011         probability_threshold=probability_threshold,
   1012         experiment_custom_tags=experiment_custom_tags,
   1013         engine=engine,
   1014         verbose=verbose,
   1015         return_train_score=return_train_score,
   1016         **kwargs,
   1017     )

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/pycaret/classification/oop.py:1341, in ClassificationExperiment.create_model(self, estimator, fold, round, cross_validation, fit_kwargs, groups, experiment_custom_tags, probability_threshold, engine, verbose, return_train_score, **kwargs)
   1338     self._set_engine(estimator=estimator, engine=engine, severity="error")
   1340 try:
-> 1341     return_values = super().create_model(
   1342         estimator=estimator,
   1343         fold=fold,
   1344         round=round,
   1345         cross_validation=cross_validation,
   1346         fit_kwargs=fit_kwargs,
   1347         groups=groups,
   1348         verbose=verbose,
   1349         experiment_custom_tags=experiment_custom_tags,
   1350         probability_threshold=probability_threshold,
   1351         return_train_score=return_train_score,
   1352         **kwargs,
   1353     )
   1354 finally:
   1355     if engine is not None:
   1356         # Reset the models back to the default engines

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/pycaret/internal/pycaret_experiment/supervised_experiment.py:1753, in _SupervisedExperiment.create_model(self, estimator, fold, round, cross_validation, predict, fit_kwargs, groups, refit, probability_threshold, experiment_custom_tags, verbose, return_train_score, **kwargs)
   1741 # TODO improve error message
   1742 assert not any(
   1743     x
   1744     in (
   (...)
   1751     for x in kwargs
   1752 )
-> 1753 return self._create_model(
   1754     estimator=estimator,
   1755     fold=fold,
   1756     round=round,
   1757     cross_validation=cross_validation,
   1758     predict=predict,
   1759     fit_kwargs=fit_kwargs,
   1760     groups=groups,
   1761     refit=refit,
   1762     probability_threshold=probability_threshold,
   1763     experiment_custom_tags=experiment_custom_tags,
   1764     verbose=verbose,
   1765     return_train_score=return_train_score,
   1766     **kwargs,
   1767 )

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/pycaret/internal/pycaret_experiment/supervised_experiment.py:1522, in _SupervisedExperiment._create_model(self, estimator, fold, round, cross_validation, predict, fit_kwargs, groups, refit, probability_threshold, experiment_custom_tags, verbose, system, add_to_model_list, X_train_data, y_train_data, metrics, display, model_only, return_train_score, **kwargs)
   1519         return model, model_fit_time
   1520     return model
-> 1522 model, model_fit_time, model_results, _ = self._create_model_with_cv(
   1523     model,
   1524     data_X,
   1525     data_y,
   1526     fit_kwargs,
   1527     round,
   1528     cv,
   1529     groups,
   1530     metrics,
   1531     refit,
   1532     system,
   1533     display,
   1534     return_train_score=return_train_score,
   1535 )
   1537 # end runtime
   1538 runtime_end = time.time()

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/pycaret/internal/pycaret_experiment/supervised_experiment.py:1114, in _SupervisedExperiment._create_model_with_cv(self, model, data_X, data_y, fit_kwargs, round, cv, groups, metrics, refit, system, display, return_train_score)
   1112 model_fit_start = time.time()
   1113 with redirect_output(self.logger):
-> 1114     scores = cross_validate(
   1115         pipeline_with_model,
   1116         data_X,
   1117         data_y,
   1118         cv=cv,
   1119         groups=groups,
   1120         scoring=metrics_dict,
   1121         fit_params=fit_kwargs,
   1122         n_jobs=n_jobs,
   1123         return_train_score=return_train_score,
   1124         error_score=0,
   1125     )
   1126 model_fit_end = time.time()
   1127 model_fit_time = np.array(model_fit_end - model_fit_start).round(2)

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/sklearn/model_selection/_validation.py:252, in cross_validate(estimator, X, y, groups, scoring, cv, n_jobs, verbose, fit_params, pre_dispatch, return_train_score, return_estimator, error_score)
     49 def cross_validate(
     50     estimator,
     51     X,
   (...)
     63     error_score=np.nan,
     64 ):
     65     """Evaluate metric(s) by cross-validation and also record fit/score times.
     66 
     67     Read more in the :ref:`User Guide <multimetric_cross_validation>`.
   (...)
    250     [0.28009951 0.3908844  0.22784907]
    251     """
--> 252     X, y, groups = indexable(X, y, groups)
    254     cv = check_cv(cv, y, classifier=is_classifier(estimator))
    256     if callable(scoring):

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/sklearn/utils/validation.py:433, in indexable(*iterables)
    414 """Make arrays indexable for cross-validation.
    415 
    416 Checks consistent length, passes through None, and ensures that everything
   (...)
    429     sparse matrix, or dataframe) or `None`.
    430 """
    432 result = [_make_indexable(X) for X in iterables]
--> 433 check_consistent_length(*result)
    434 return result

File /opt/tljh/user/envs/pycaret-git/lib/python3.9/site-packages/sklearn/utils/validation.py:387, in check_consistent_length(*arrays)
    385 uniques = np.unique(lengths)
    386 if len(uniques) > 1:
--> 387     raise ValueError(
    388         "Found input variables with inconsistent numbers of samples: %r"
    389         % [int(l) for l in lengths]
    390     )

ValueError: Found input variables with inconsistent numbers of samples: [15959, 15959, 22800]
```

### Installed Versions

<details>
'3.0.0.rc4'
</details>

**Comment by Ali-Flt** [comment]
Have you tried giving `data["EDUCATION"]` directly as the setup argument?
```
fold_groups=data["EDUCATION"]
```

**Comment by ninjit** [comment]
... that's what I did and described in the problem statement and code
```Python
...
data_groups = data["EDUCATION"]
...
... ..., fold_groups=data_groups,... ...
```

**Comment by Ali-Flt** [comment]
I guess they are basically the same, I said that because in my code I did it the way I mentioned and it works (although I have no Idea if it's really working properly or just not returning errors)
I think groupKfold has many issues in the current version which they are addressing in [rc5](https://github.com/pycaret/pycaret/milestone/18)

**Comment by tvdboom** [comment]
Hi, thanks for noticing the bug. I fixed it in #3057. The fix will be available with next release

**Comment by Ali-Flt** [comment]
@tvdboom thanks for the fix, Is this PR also going to fix any of #3034 or #2960?

**Comment by tvdboom** [comment]
Yes, hadn't seen those issues but I ran a test for each and it seems to work now
